orangecontrib/geo/utils.py

```diff
@@ -6,45 +6,61 @@
 from Orange.data.domain import filter_visible
 
 
-def find_lat_lon(data, filter_hidden=False, fallback=True):
-    """Return inferred latitude and longitude attributes as found in the data domain"""
+LATITUDE_NAMES = ('latitude', 'lat')
+LONGITUDE_NAMES = ('longitude', 'lng', 'long', 'lon')
+
+
+def find_lat_lon(data, filter_hidden=False):
+    """
+    Infer latitude and longitude attributes from data
+
+    - If there are less than two numeric variables, return None, None
+    - If there are variables with recognized names (see LATITUDE_NAMES,
+      LONGITUDE_NAMES), return them
+    - Otherwise, if there are exactly two numeric variables
+       - if one has a matching name, the other is chosen to be the other coord
+       - if one has all values below 90 and the other has at least one above,
```

Comment:
Exactly. If there are two numeric variables, we try to determine which is which. One must have values between -90 and 90 and the other between -180 and 180 (including some that are not between -90 and 90). If this is the case, we know which is which. If not, we take (the "otherwise" case) the first one to be latitude and the other to be longitude.

See the test (https://github.com/biolab/orange3-geo/pull/146/commits/548ab9fd44a68ea784e8e38bee0e799f9d2a8250#diff-9096bfbbbdb40d1dd0d946e0d4debef674af24060f72f44e4c6a68628d2d7debR69): the variable with values 12 and 45 is latitude and the one with 8 and 120 is longitude -- it cannot be the opposite.

One of the following tests has a situation in which both are -90 and 90, so it can't decide and just uses the order,

Reply:
If it comes to `for`, the function returns either `lat_attr, lon_attr = cont_vars[::-1]` or , `lat_attr, lon_attr = cont_vars`, whichever fulfills the criteria that the maximal value of the first is below and of the second above 90. So it does not (at least I believe so) return the first variable twice.

You are right, though, that I didn't put the limit on longitude. Do I understand correctly your comment that I should  add `<= 180`?

Reply:
Confirmed. I was testing, stupidly, with more than 2 numeric variables. The behaviour is correct.

Yes, I think the limit for longitude should be <=180 otherwise strange variables can be recognised as lon. That might get tricky for large negatives (-200?), but let's leave it as is for now.